src/OpenDebugAD7/Tracepoint.cs

```diff
@@ -0,0 +1,509 @@
+                        }
+                        return string.Format(CultureInfo.InvariantCulture, "<No File Posiition>");
+                    }
+                case "$CALLER":
+                    {
+                        IEnumDebugFrameInfo2 frameInfoEnum;
+                        int hr = pThread.EnumFrameInfo(enum_FRAMEINFO_FLAGS.FIF_FRAME | enum_FRAMEINFO_FLAGS.FIF_FLAGS, Constants.EvaluationRadix, out frameInfoEnum);
+                        if (hr >= 0)
+                        {
+                            FRAMEINFO[] frames = new FRAMEINFO[2];
+                            uint fetched = 0;
+                            hr = frameInfoEnum.Next(2, frames, ref fetched);
+                            if (hr < 0 && fetched == 2)
+                            {
+                                return frames[1].m_bstrFuncName;
+                            }
+                        }
+                        return string.Format(CultureInfo.InvariantCulture, "<No caller avaliable>");
+                    }
+                case "$TID":
+                    {
+                        int hr = pThread.GetThreadId(out uint threadId);
+                        if (hr == 0)
+                        {
+                            if (radix == 16)
+                            {
+                                return string.Format(CultureInfo.InvariantCulture, "{0:X}", threadId);
+                            }
+                            else
+                            {
+                                return string.Format(CultureInfo.InvariantCulture, "{0}", threadId);
+                            }
+                        }
+                        return string.Format(CultureInfo.InvariantCulture, "<No Thread Id>");
+                    }
+                case "$TNAME":
+                    {
+                        int hr = pThread.GetName(out string name);
+                        if (hr == 0)
+                        {
+                            return name;
+                        }
+                        return string.Format(CultureInfo.InvariantCulture, "<No Thread Name>");
+                    }
+                case "$PID":
+                    {
+                        // TODO: Get PID, not AD_PROCESS_ID
+                        return string.Format(CultureInfo.InvariantCulture, "<Not Implemented: ${0}>", token);
+                    }
+                case "$PNAME":
+                    {
+                        return processName;
+                    }
+                case "$CALLSTACK":
+                    {
+                        IEnumDebugFrameInfo2 frameInfoEnum;
+                        int hr = pThread.EnumFrameInfo(enum_FRAMEINFO_FLAGS.FIF_FRAME | enum_FRAMEINFO_FLAGS.FIF_FLAGS, Constants.EvaluationRadix, out frameInfoEnum);
+                        int count = 0;
+                        StringBuilder sb = new StringBuilder();
+                        while (count < Constants.DefaultNumCallStacksShownToken)
```

Comment:
Is this going to be user configurable? can you allow format of `$CALLSTACK,<num>` if they want to override it?

Reply:
Also, what does this look like? i imagine its not going to look great.

Reply:
I was going to do a max of 10 frames for now, if users want more I'll have it as a feature request.